### PR TITLE
gnome.zenity: relax meta.platforms

### PR DESCRIPTION
**Pull request by abathur**
###### Description of changes

- Relax the platforms for `gnome.zenity` from `linux` to `unix`. I'm not certain if relaxing it _this_ far is correct, but I have `allowUnsupportedSystem = true` set on macOS and didn't even realize this was marked Linux-only until now. Screenshot:

    Screen Shot 2022-12-08 at 8 50 50 PM

- Marked `tev` unbuildable on darwin. It has build problems that were shielded by zenity not being buildable. See https://github.com/NixOS/nixpkgs/pull/205247#issuecomment-1344522245 for a little more context.
- Updated `luaPackages.nfd` to build on darwin. It also had build problems exposed by zenity being buildable, but the fix was straightforward.

Reviews for the full PR:

Result of `nixpkgs-review` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>lua52Packages.nfd</li>
    <li>lua53Packages.nfd</li>
    <li>tev</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>gnome.zenity</li>
    <li>gnomeExtensions.ddterm</li>
    <li>kodi-cli</li>
    <li>lua51Packages.nfd</li>
    <li>luajitPackages.nfd</li>
    <li>vimPlugins.vCoolor-vim</li>
  </ul>
</details>

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by abathur** [comment]
@scoder12 Figured I should give you a heads up on marking `nfd` (but not trying to get you to try to fix it).

File: pkgs/applications/graphics/tev/default.nix [hunk 1]
```diff
@@ -49,6 +49,7 @@ stdenv.mkDerivation rec {
     license = licenses.bsd3;
     platforms = platforms.unix;
     badPlatforms = [ "aarch64-linux" ]; # fails on Hydra since forever
+    broken = stdenv.isDarwin; # builds need work
```
Comment (Atemu) [inline pkgs/applications/graphics/tev/default.nix:52]: Nit: Could you be a little more specific what the error is?

File: pkgs/development/lua-modules/nfd/default.nix [hunk 2]
```diff
@@ -36,6 +36,7 @@ buildLuarocksPackage {
       "A tiny, neat lua library that portably invokes native file open and save dialogs.";
     homepage = "https://github.com/Alloyed/nativefiledialog/tree/master/lua";
     license = lib.licenses.zlib;
+    broken = stdenv.isDarwin; # builds need work
```
Comment (Atemu) [inline pkgs/development/lua-modules/nfd/default.nix:39]: Here too.

File: pkgs/applications/graphics/tev/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/applications/graphics/tev/default.nix:52]: the comment should at least indicate what part and a short extract from the error. Right now it is not very helpfull.

File: pkgs/development/lua-modules/nfd/default.nix (hunk 2 above)
Comment (SuperSandro2000) [inline pkgs/development/lua-modules/nfd/default.nix:39]: same here

**Review by Atemu — APPROVED** [review]
Result of `nixpkgs-review pr 205247` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gnome.zenity</li>
  </ul>
</details>

Runs for me.

**Comment by abathur** [comment]
I've pushed an update that gets `nfd` building (and expands the comment on `tev`).

Regarding trying to build `tev`, adopting wegank's decision got this most of the way, though it still eventually runs into the error below. Light searching suggests it's an SDK error. I can't use `apple_sdk_11_0` on this system, so that's where the path ends for me. I've pushed up the WIP in https://github.com/abathur/nixpkgs/commit/f828b65015861578fa8c7f8886b0173496ca2b2d and updated

<details><summary>no matching function error; SDK?</summary>

```console
[ 92%] Building CXX object CMakeFiles/tev.dir/src/FalseColor.cpp.o
/tmp/nix-build-tev-1.23.drv-0/source/src/Common.cpp:59:12: error: no viable conversion from returned value of type 'std::__1::u8string' (aka 'basic_string<char8_t>') to function return type 'fs::path'
    return toU8string(utf8);
           ^~~~~~~~~~~~~~~~
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/filesystem:763:29: note: candidate constructor not viable: no known conversion from 'std::__1::u8string' (aka 'basic_string<char8_t>') to 'const std::__1::__fs::filesystem::path &' for 1st argument
  _LIBCPP_INLINE_VISIBILITY path(const path& __p) : __pn_(__p.__pn_) {}
                            ^
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/filesystem:764:29: note: candidate constructor not viable: no known conversion from 'std::__1::u8string' (aka 'basic_string<char8_t>') to 'std::__1::__fs::filesystem::path &&' for 1st argument
  _LIBCPP_INLINE_VISIBILITY path(path&& __p) noexcept
                            ^
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/filesystem:768:3: note: candidate constructor not viable: no known conversion from 'std::__1::u8string' (aka 'basic_string<char8_t>') to 'std::__1::__fs::filesystem::path::string_type &&' (aka 'basic_string<char> &&') for 1st argument
  path(string_type&& __s, format = format::auto_format) noexcept
  ^
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/filesystem:772:3: note: candidate template ignored: requirement '__is_pathable<std::__1::basic_string<char8_t, std::__1::char_traits<char8_t>, std::__1::allocator<char8_t>>, false, false, false>::value' was not satisfied [with _Source = std::__1::basic_string<char8_t, std::__1::char_traits<char8_t>, std::__1::allocator<char8_t>>]
  path(const _Source& __src, format = format::auto_format) {
  ^
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/string:875:5: note: candidate function
    operator __self_view() const _NOEXCEPT { return __self_view(data(), size()); }
    ^
/tmp/nix-build-tev-1.23.drv-0/source/src/Common.cpp:66:12: error: no matching function for call to 'fromU8string'
    return fromU8string(path.u8string());
           ^~~~~~~~~~~~
/tmp/nix-build-tev-1.23.drv-0/source/src/Common.cpp:35:8: note: candidate function not viable: no known conversion from 'basic_string<char, char_traits<char>, allocator<char>>' to 'const basic_string<char8_t, char_traits<char8_t>, allocator<char8_t>>' for 1st argument
string fromU8string(const u8string& str) {
       ^
[ 92%] Building CXX object CMakeFiles/tev.dir/src/HelpWindow.cpp.o
2 errors generated.
make[2]: *** [CMakeFiles/tev.dir/build.make:280: CMakeFiles/tev.dir/src/Common.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:558: CMakeFiles/tev.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
error: builder for '/nix/store/d5jaj5br8pna7jsgb41a5h6f5hw072wm-tev-1.23.drv' failed with exit code 2
```
</details>

File: pkgs/development/lua-modules/nfd/default.nix [hunk 3]
```diff
@@ -1,5 +1,5 @@
-{ fetchFromGitHub, buildLuarocksPackage, lua, pkg-config, lib
-, substituteAll, zenity }:
+{ stdenv, fetchFromGitHub, buildLuarocksPackage, lua, pkg-config, lib
+, substituteAll, zenity, darwin }:
```
Comment (wegank) [inline pkgs/development/lua-modules/nfd/default.nix:2]:
```suggestion
, substituteAll, zenity, AppKit }:
```

then, add `inherit (darwin.apple_sdk.frameworks) AppKit;` in `all-packages.nix`.

Comment (abathur) [reply pkgs/development/lua-modules/nfd/default.nix]: Force-pushed to adopt.

File: pkgs/development/lua-modules/nfd/default.nix [hunk 4]
```diff
@@ -25,6 +25,9 @@ buildLuarocksPackage {
   extraVariables.LUA_LIBDIR = "${lua}/lib";
   nativeBuildInputs = [ pkg-config ];
 
+  buildInputs = [ ]
+    ++ lib.optionals stdenv.isDarwin [ AppKit ];
+
```
Comment (Atemu) [inline pkgs/development/lua-modules/nfd/default.nix:30]: Nit: This pattern is a bit odd.

**Review by Atemu — APPROVED** [review]
Result of `nixpkgs-review pr 205247` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>lua52Packages.nfd</li>
    <li>lua53Packages.nfd</li>
    <li>luaPackages.nfd</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>gnome.zenity</li>
    <li>lua51Packages.nfd</li>
    <li>luajitPackages.nfd</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

File: pkgs/development/lua-modules/nfd/default.nix [hunk 5]
```diff
@@ -25,6 +25,9 @@ buildLuarocksPackage {
   extraVariables.LUA_LIBDIR = "${lua}/lib";
   nativeBuildInputs = [ pkg-config ];
 
+  buildInputs = [ ]
+    ++ lib.optionals stdenv.isDarwin [ AppKit ];
```
Comment (SuperSandro2000) [inline pkgs/development/lua-modules/nfd/default.nix:29]:
```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [ AppKit ];
```

**Review by spencerpogo — APPROVED** [review]

**Pull request merged**